### ISSUE TITLE
Allow cloud.id and cloud.auth in filebeat.yml

### ISSUE THREAD

**Issue by Luckless** [pcfens/puppet-filebeat#266]
Would like the filebeat.yml file to allow the cloud.id and cloud.auth for using elastic.co hosted solution easily

**Comment by Luckless** [comment]
Testing this out with my elastic.co POC

**Comment by pcfens** [comment]
This sounds like a great idea.

To match the existing structure I'd imagine that the parameter might be a hash named cloud so that it looks like
```puppet
class { '::filebeat':
  cloud => {
    id    => '123456',
    auth  => 'abcdef',
  }
...
}
```

that way we'll also be ready should other options be required in the future.

PRs are always great if you have time to work on it, and thanks for the idea!

**Comment by Luckless** [comment]
Yeah, I tried that but it expects `cloud.id` and `cloud.auth`, not in a hash. I think I have my commit in a good place but I have to work through all the networking stuff to get filebeats to connect through my VPC

I'll open a PR when I'm sure. Current setup is optional

```
filebeat::cloud_id 
filebeat::cloud_auth
```

**Comment by pcfens** [comment]
Have you tried with the YAML looking like
```yaml
cloud:
  id: 123456
  auth: abcdef
```

Typically dots in a yaml key indicate a hash, even if examples don't show it working that way.

**Comment by pcfens** [comment]
Fixed by #267
